src/WebJobs.Script/StorageProvider/HostAzureTableStorageProvider.cs

```diff
@@ -0,0 +1,60 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Net.Http;
+using Azure.Core.Pipeline;
+using Azure.Data.Tables;
+using Microsoft.Extensions.Azure;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.Azure.WebJobs.Script
+{
+    internal class HostAzureTableStorageProvider : IAzureTableStorageProvider
+    {
+        private readonly ILogger<HostAzureTableStorageProvider> _logger;
+        private readonly IDelegatingHandlerProvider _delegatingHandlerProvider;
+        private readonly TableServiceClientProvider _tableServiceClientProvider;
+
+        public HostAzureTableStorageProvider(IConfiguration configuration, ILogger<HostAzureTableStorageProvider> logger, AzureComponentFactory componentFactory, AzureEventSourceLogForwarder logForwarder, IDelegatingHandlerProvider delegatingHandlerProvider)
+        {
+            _logger = logger ?? throw new ArgumentNullException(nameof(logger));
+            _delegatingHandlerProvider = delegatingHandlerProvider;
+            _tableServiceClientProvider = new TableServiceClientProvider(componentFactory, logForwarder);
+            Configuration = configuration ?? throw new ArgumentNullException(nameof(configuration));
+        }
+
+        public virtual IConfiguration Configuration { get; private set; }
```

Comment:
Should this need to be a public property? Looks like it's been used only inside the class. So we could keep it as a private variable. Remove it from the interface as well?

Reply:
Indeed not for our current usage. I based this provider in its equivalent for `BlobStorage`: `HostAzureTableStorageProvider` where the interface comes from `WebJobs.Host.Storage`. Let me apply your suggestions and we can always check in the future if there's a need to move this to the WebJobs layer and/or increase/modify the public surface.